### ISSUE TITLE
Can't Upgrade To V3 from V2.1.1

### ISSUE THREAD

**Issue by OfficialKaos** [hoobs-org/HOOBS#357]
**Description**
Homebridge to Hoobs migration tool won't work

**Version**
2.1.1
**Did you upgrade**
Please let us know if you upgraded from a previous version.
- [ ] Yes
- [x] No

**Pervious version**
If you upgraded, please let us know your previous version.

**Did you orginally upgrade to HOOBS 3 from HOOBS 2.1.1?**
Did you perform an upgrade from HOOBS 2.1.1? Some things are different.
- [ ] Yes
- [ ] No

**What device are you using?**
Please let us know the device you are running HOOBS on.
- [ ] HOOBS Box
- [x] HOOBS SD Card
- [ ] Custom Install

**If this is a custom install, what's your operating system?**
Please let us know what operating system and version you are using. Ex. Fedora 30 or macOS Catalina.

**List your plugins**
Please include a list of the plugins you are using. We need to be able to find it on NPM. If the plugin is no published, please include a link to the repository.

**Post your config**
This can be found in the interface, Configuration -> Advanced or you can get it via SSH `cat ~/.hoobs/etc/config.json`

* Remove any passwords from the config file. This is public.

```
POST CONFIG HERE
```

**Post your log**
You can get the log from the intercafe. This is in the Log section.

```
POST LOG HERE
```

**Additional context**

**Comment by OfficialKaos** [comment]
This is the error message
USER: hoobs
DIR: /usr/local/lib
CMD: sudo -E -n npm --no-update-notifier install --unsafe-perm homebridge-to-hoobs@latest

sudo: npm: command not found

Command failed. Please review log for details.

**Comment by mkellsy** [comment]
The no NPM error is coming from UI-X. Can you open the terminal and type this.
```
npm -v
```

What does it say?

**Comment by OfficialKaos** [comment]
hoobs@hoobs:/var/lib/homebridge $ npm -v
bash: npm: command not found
hoobs@hoobs:/var/lib/homebridge $

**Comment by OfficialKaos** [comment]
I have several hombridge instances running 2.1.1,  UI-X 4.6.3 and 4.6.6 both return same error message as above 
4.6.5 says 6.9.0

**Comment by mkellsy** [comment]
OK. It looks like Node is corrupt on some of your devices. I can direct you on how to fix it, but I need you to run this first.

```
uname -m
```

It will tell me your architecture (arm6l or arm7l). Then I can tell you what to run and how to fix it.

**Comment by OfficialKaos** [comment]
Thanks, there all armv7l

**Comment by mkellsy** [comment]
I am not sure how Node got corrupt, but I am guessing it was the 2.1.1 hoobs-upgrade command. This was from before my time with HOOBS.

Here's how to fix;

1. Update Raspbian repos
```
sudo apt-get update
```

2. Make sure Curl, Tar and Git are installed
```
sudo apt-get install -y curl tar git
``

3. Re-install Node
```
curl -O https://nodejs.org/dist/v10.17.0/node-v10.17.0-linux-armv7l.tar.gz
sudo tar -xzf ./node-v10.17.0-linux-armv7l.tar.gz -C /usr/local --strip-components=1 --no-same-owner
rm -f ./node-v10.17.0-linux-armv7l.tar.gz
```

4. Set the NPM prefix
```
sudo npm config set -g prefix /usr/local
```

5. If you are still running 2.1.1, you need to reinstall homebridge-config-ui-x
```
sudo npm uninstall -g homebridge-config-ui-x
sudo npm install -g --unsafe-perm homebridge-config-ui-x
```

**Comment by OfficialKaos** [comment]
ok done with no errors .. should i restart the pi and try the hombridge to hoobs plugin again now?

**Comment by mkellsy** [comment]
Yep. Did you see your other issue. I want to look at your other box.

**Comment by mkellsy** [comment]
This had a wired NPM prefix set. Here's how I found out and what I did to fix it.

1. NPM was installing, but none of the binaries were found, so hoobs-init wouldn't run.
2. Check the NPM prefix

```
npm get prefix
```

It should be either `/usr` or `/usr/local`

3. Locate the global node_modules
```
ls -la /usr/local/lib
```
or
```
ls -la /usr/lib
```

In this case the global node_modules were in `/usr/local`

4. Set the proper prefix
```
sudo npm config set -g prefix /usr/local
```

5. Delete the improper modules and bin directories. In this case it was in `/usr/local/lib/node_modules/lib`
```
sudo rm -fR /usr/local/lib/node_modules/lib
sudo rm -fR /usr/local/lib/node_modules/bin
```

6. Check NPM
```
npm -v
```

If not found re-install Node, in this case it was OK

7. Attempt to upgrade.
